File: funcscript/parser/syntax/FuncScriptParser.Parse.cs
using FuncScript.Block;
using FuncScript.Funcs.Math;
using FuncScript.Model;

namespace FuncScript.Core
{
    public partial class FuncScriptParser
    {
        public static ExpressionBlockResult Parse(ParseContext context)
        {
            return GetRootExpression(context);
        }
        public static ExpressionBlockResult Parse(string expression)
        {
            var context = new ParseContext
            (
                new DefaultFsDataProvider(),
                expression,
                new List<SyntaxErrorData>()
            );
            return GetRootExpression(context);
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.Parse.ts
import { ParseContext, ParseResult } from "../FuncScriptParser.Main"; 
import { GetRootExpression } from "./FuncScriptParser.GetRootExpression";

export function Parse(context: ParseContext): ParseResult {
    const rootExpressionResult = GetRootExpression(context);
    return rootExpressionResult;
}